File: sdk/cpp/src/api/network.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/network.hpp>

namespace hypermesh {

PeerList NetworkApi::peers() const {
    auto j = http_.get("/api/v1/network/peers");
    return j.get<PeerList>();
}

} // namespace hypermesh

File: sdk/typescript/src/api/network.ts
import type { HttpClient } from "../client.js";
import type { PeersResponse } from "../types.js";

export class NetworkApi {
  constructor(private readonly http: HttpClient) {}

  async peers(): Promise<PeersResponse> {
    return this.http.get<PeersResponse>("/api/v1/network/peers");
  }
}
